javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,127 @@
+let personName, nameIsSaved = false,
+  processedText, response, todoList = [];
+
+function getReply(command) {
+  processedText = processText(command);
+  //  console.log(processedText);
+  if (processedText.substring(0, 13) == "hellomynameis" && !nameIsSaved) {
+    nameIsSaved = true;
+    personName = command.substring(17, command.length);
+    return ("Nice to meet you " + personName);
+  } else if (processedText.substring(0, 13) == "hellomynameis" && nameIsSaved) {
+    return ("You already provided your name! Aren't you " + personName + "?");
+  } else if (processedText.substring(0, 3) == "add") {
+    todoList.push(command.substring(4, command.length - 11));
+    return (command.substring(4, command.length - 11) + " added to your todo");
+  } else if (processedText.substring(0, 6) == "remove") {
+    todoList = todoList.filter(e => e !== command.substring(7, command.length - 13));
+    return (command.substring(7, command.length - 13) + " removed from your todo");
+  } else if (processedText.includes("+") || processedText.includes("-") || processedText.includes("*") || processedText.includes("/")) {
+    let math = processedText.replace("whatis", "");
+    math = math.replace("?", "");
+    return (math + " = " + eval(math));
+  } else if (processedText.substring(4, 9) == "timer") {
+    let timerMin = parseInt(processedText.substring(12, 14), 10);
+    startTimer(timerMin);
+    return ("Timer: is set for " + timerMin + " minutes");
+  } else if (processedText.includes("quote")) {
+    quoteOfTheDay()
+    console.log("quote is found");
+  } else {
+    return processCommand(processedText);
+  }
+}
+
+function processText(rawText) {
+  let text = rawText.replace(/\s/g, "");
+  text = text.toLowerCase();
+  return text;
+}
+
+function startTimer(min) {
+  var sec = min * 60;
+  var timer = setInterval(function() {
+    sec--;
+    if (sec < 0) {
+      clearInterval(timer);
+      console.log("Timer: the time is up!");
+    }
+  }, 1000);
+}
+
+function quoteOfTheDay() {
+  var xmlhttp = new XMLHttpRequest();
+  xmlhttp.onreadystatechange = function() {
+    if (this.readyState == 4 && this.status == 200) {
+      let myObj = JSON.parse(this.responseText);
+      alert(myObj.name);
+      console.log(myObj.name);
+    }
+  };
+  xmlhttp.open("GET", "https://www.w3schools.com/js/json_demo.txt", true);
+  xmlhttp.send();
+
+}
+
+function processCommand(cleanText) {
+  switch (cleanText) {
+    case "whatismyname?":
+      response = "Your name is " + personName + "!";
+      break;
+    case "whatisonmytodo?": // "you have 2 todos - fishing and singing in the shower"
+      let todoListItems = "";
+      for (var i = 0; i < todoList.length - 1; i++) {
+        todoListItems += todoList[i];
+        todoListItems += " and ";
+      }
```

Comment:
You can simplify this using `.join`:
```js
todoListItems = todoList.join(" and ");
```
This will add " and " between all items, but not before the first one or after the last one.